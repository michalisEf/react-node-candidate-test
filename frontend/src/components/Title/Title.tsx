import React from 'react';

import './Title.css';
import { ITitle } from '../../interfaces';

const Title = ({ title }: ITitle): JSX.Element => {
	return (
		<h1 className="title">{ title }</h1>
	);
};

export default Title;
