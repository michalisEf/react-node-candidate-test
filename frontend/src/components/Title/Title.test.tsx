import React from 'react';
import { render, screen } from '@testing-library/react'

import Title from './Title';

it('renders Title component', () => {
	const props = {
		title: 'React Developer in London Â£90K',
	};
	render(<Title { ...props }/>);
	const title = screen.getByText(/React Developer/)
	expect(title).toBeInTheDocument();
	expect(title).toMatchSnapshot();
});
