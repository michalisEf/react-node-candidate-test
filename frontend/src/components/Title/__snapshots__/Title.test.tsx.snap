// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Title component 1`] = `
<h1
  class="title"
>
  React Developer in London Â£90K
</h1>
`;
